### DISCUSSION TITLE
Typed `TaskGroup.start_soon`

### DISCUSSION THREAD

**Discussion by notypecheck** [agronholm/anyio#611] [Ideas]
Would it be possible to provide asyncio-like interface to `TaskGroup.start_soon`?:
```py
tg.start_soon(coro(a=a, b=b))
```
Instead of 
```py
tg.start_soon(coro, a, b)
```

If that's not possible due to `trio` API maybe it should at least be typed using `ParamSpec`? 🤔

**Comment by agronholm** [comment]
1. Trio does not support this
2. This would not work with `TaskGroup.start()` anyway

**Comment by smurfix** [comment]
That's not a good idea because, frankly, the asyncio way is broken. For instance, if `coro` has a non-async wrapper (debugging, change the floating-point context, …) the wrapper doesn't run in the new task's context. Oops.

If you want to call a coro with keyword parameters, `functools.partial` is your friend.

**Reply by notypecheck** [reply]
Shouldn't you create your own wrapper for `functools.partial` since it's not typed? 🤔 Also it's not about keyword parameters but typing/type safety in general

**Reply by davidbrochart** [reply]
Maybe worth looking at: https://github.com/tiangolo/asyncer

**Reply by agronholm** [reply]
You can often also use a `lambda` too (but be careful about its side effects).

**Reply by notypecheck** [reply]
Thanks for quick replies, I think making a wrapper is the easiest way to avoid any issues with typing here
